crucible-mir/src/Mir/Intrinsics.hs

```diff
@@ -1184,195 +1182,200 @@ dropRefRoot _bak _gs (Const_RefRoot tpr _) =
     leafAbort $ GenericSimError $
         "Cannot drop Const_RefRoot (of type " ++ show tpr ++ ")"
 
+typedLeafOp ::
```

Comment:
Could you leave a Haddock comment on what this function does? It's not entirely obvious at a glance.